src/models/frame.ts

```diff
@@ -39,57 +70,55 @@ export class FrameNavigation {
  */
 export class FrameStore {
   private frames: { [frameId: string]: FrameNavigation };
-  private loadResolver: () => void;
 
   constructor() {
     this.frames = {};
   }
-  start(frameId: string, waitForLoad: boolean = false): Promise<PageNavigateResult> {
-    const nav = new FrameNavigation();
+  start(frameId: string, waitForLoad: boolean): Promise<Network.ResponseReceivedParameters[]> {
+    const nav = new FrameNavigation(waitForLoad);
     this.frames[frameId] = nav;
     const navPromise = nav.getPromise();
-    return waitForLoad ?
-      Promise.all([navPromise, new Promise((r) => { this.loadResolver = r; })]).then(([n]) => n) :
-      navPromise;
+    return navPromise;
   }
   onNetworkResponse(res: Network.ResponseReceivedParameters) {
     if (res.frameId) {
       const nav = this.frames[res.frameId];
-      // TODO: Figure out why sometimes we recieve a bogus requestId that is a
-      // decimal number represented as a string
-      if (nav && res.requestId.indexOf('.') < 0) {
+      if (nav) {
         nav.onNetworkResponse(res);
       }
     } else {
       throw new Error('Received network response without frameId');
     }
   }
+  onRequestWillBeSent(req: Network.RequestWillBeSentParameters) {
+    if (req.frameId) {
+      const nav = this.frames[req.frameId];
+      if (nav) {
+        nav.onRequestWillBeSent(req);
+      }
+    } else {
+      throw new Error('Received network request will be sent without frameId');
+    }
+  }
   onNavigationComplete(result: Page.FrameNavigatedParameters) {
     const nav = this.frames[result.frame.id];
     if (nav) {
-      nav.onNavigationComplete(result);
+      nav.onNavigationComplete();
     }
   }
+
   onLoadEvent() {
-    if (this.loadResolver) {
-      this.loadResolver();
-    }
+    Object.keys(this.frames).forEach((frameId) => {
+      this.frames[frameId].onLoadEvent();
+    });
   }
 }
 
-/**
- * Represents the result of a navigation
- * @public
- */
-export interface PageNavigateResult {
-  networkResult: Network.ResponseReceivedParameters;
-  frame: Page.Frame;
+export interface ResponseHash {
```

Comment:
Instead of returning a `ResponseHash` from `navigate`, why not keep returning `PageNavigateResult` and then change `PageNavigateResult` to be something like:
```TypeScript
export interface PageNavigateResult {
   page: CapturedResponse;
   childResponses: ResponseHash;
   frame: Page.Frame;
}
```
This also eliminates the need to expose the acceptance utils.